pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -364,3 +364,53 @@ func (qt *QueueTracker) UnlinkQT(queuePath string) bool {
 	}
 	return false
 }
+
+func (qt *QueueTracker) canRunApp(queuePath string, applicationID string, trackType trackingType) bool {
+	log.Log(log.SchedUGM).Debug("Checking can run app",
+		zap.Int("tracking type", int(trackType)),
+		zap.String("queue path", queuePath),
+		zap.String("application", applicationID))
+	childQueuePath, immediateChildQueueName := getChildQueuePath(queuePath)
+	if childQueuePath != common.Empty {
+		if qt.childQueueTrackers[immediateChildQueueName] != nil {
+			allowed := qt.childQueueTrackers[immediateChildQueueName].canRunApp(childQueuePath, applicationID, trackType)
+			if !allowed {
+				return false
+			}
+		}
+	}
+
+	running := len(qt.runningApplications)
+	if existingApp := qt.runningApplications[applicationID]; !existingApp {
+		running += 1
```

Comment:
Can we return when app is already running? so that rest of the following code would execute only for new application.